nomad/acl.go

```diff
@@ -185,27 +293,28 @@ func (s *Server) ResolveSecretToken(secretID string) (*structs.ACLToken, error)
 	}
 	defer metrics.MeasureSince([]string{"nomad", "acl", "resolveSecretToken"}, time.Now())
 
+	if secretID == "" {
+		return structs.AnonymousACLToken, nil
+	}
+	if !helper.IsUUID(secretID) {
+		return nil, structs.ErrTokenInvalid
+	}
+
 	snap, err := s.fsm.State().Snapshot()
 	if err != nil {
 		return nil, err
 	}
 
 	// Lookup the ACL Token
-	var token *structs.ACLToken
-	// Handle anonymous requests
-	if secretID == "" {
-		token = structs.AnonymousACLToken
-	} else {
-		token, err = snap.ACLTokenBySecretID(nil, secretID)
-		if err != nil {
-			return nil, err
-		}
-		if token == nil {
-			return nil, structs.ErrTokenNotFound
-		}
-		if token.IsExpired(time.Now().UTC()) {
-			return nil, structs.ErrTokenExpired
-		}
+	token, err := snap.ACLTokenBySecretID(nil, secretID)
+	if err != nil {
+		return nil, err
+	}
+	if token == nil {
+		return nil, structs.ErrTokenNotFound
+	}
+	if token.IsExpired(time.Now().UTC()) {
```

Comment:
Just to nitpick - it'd be nice to get away from calling `time.Now` directly since it makes testing more clumsy; alternatives could be passing in "now" as a parameter, or turn the function into a method and have like a libtime.Clock as a struct field which can then be mocked

Reply:
Yeah I like the idea of having a mockable clock. This is moved code rather than new code in this PR though, so I'll look at returning to that in a separate PR if that works for you?